optuna/visualization/contour.py

```diff
@@ -8,15 +8,14 @@
 from optuna.visualization.utils import _is_log_scale
 from optuna.visualization.utils import is_available
 
-if type_checking.TYPE_CHECKING:
-    from typing import List  # NOQA
-    from typing import Optional  # NOQA
-    from typing import Tuple  # NOQA
+from typing import List  # NOQA
+from typing import Optional  # NOQA
+from typing import Tuple  # NOQA
 
-    from optuna.study import Study  # NOQA
-    from optuna.trial import FrozenTrial  # NOQA
-    from optuna.visualization.plotly_imports import Contour  # NOQA
-    from optuna.visualization.plotly_imports import Scatter  # NOQA
+from optuna.study import Study  # NOQA
+from optuna.trial import FrozenTrial  # NOQA
+from optuna.visualization.plotly_imports import Contour  # NOQA
+from optuna.visualization.plotly_imports import Scatter  # NOQA
```

Comment:
Excuse me, but what do you mean by "for corresponding type hints, use strings" ?

Reply:
I interpreted your comment to mean "replace a specific type specification with str" and rewrote the source code. However it doesn't still pass the mypy test......

Reply:
You can write type hints using code or strings. For instance, both of the following are valid `mypy` type hints.

```python
def _get_contour_plot(study, params: Optional[List[str]] = None) -> "go.Figure"
```

```python
def _get_contour_plot(study, params: Optional[List[str]] = None) -> go.Figure
```

The former requires `go` to be imported only when Optuna is imported via `mypy` and type checking is performed (`type_checking.TYPE_CHECKING` is true). In this case, for regular use, `go` doesn't have to be imported. The latter requires `go` to be imported in both cases. Since e.g. `Contour` may not be available during regular use, you might want to try the former.

Reply:
All right. For now, I'll try to follow your advice.